### PR TITLE
Add default directories to the $OS_BUILD_ENV_PRESERVE path

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Users of `hack/env` almost always want to be retrieving build artifacts
and script output artifacts from whatever process they trigger in the
container. Instructing every user to set a `$OS_BUILD_ENV_PRESERVE`
field is tedious, so we can default it when it is unset to contain the
normal paths for binaries, releases, RPMs and script output.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

@smarterclayton PTAL

[test]

### CODE REVIEW

File: hack/env [hunk 1]
```diff
@@ -33,4 +33,10 @@
 # NOTE:   only committed code is built.
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+relative_bin_path="$( os::util::repository_relative_path "${OS_OUTPUT_BINPATH}" )"
```
Comment (smarterclayton) [inline hack/env]: Remember this is now going to download 1GB everytime you exit a container, so just be sure you want that pain.

Comment (stevekuznetsov) [inline hack/env]: 99% sure we do ... realistically people are using this to do builds locally pretty much 99.99999% of the time, they're not running other stuff as much.

**Comment by stevekuznetsov** [comment]
[merge]

**Comment by kargakis** [comment]
An option to opt-out would be nice.

**Comment by stevekuznetsov** [comment]
This only defaults if unset -- so `OS_BUILD_ENV_PRESERVE='' hack/env` to opt out

**Comment by kargakis** [comment]
ack

**Comment by stevekuznetsov** [comment]
@openshift-bot, the [last build](https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/907/) failed from the following flakes:
 - [integration](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_integration/2869/consoleFull#165941815156bf4006e4b05b79524e5923) failed on: #8502 

re[merge]

**Pull request merged**